mltsp/tests/test_predict.py

```diff
@@ -60,3 +60,106 @@ def test_featurize_single():
                                      meta_feats)
     assert all("std_err" in d["features_dict"] for fname, d in res_dict.items())
     assert all("ts_data" in d for fname, d in res_dict.items())
+
+
+def test_featurize_tsdata():
+    """Test featurize_tsdata"""
+    res_dict = pred.featurize_tsdata(
+        os.path.join(os.path.dirname(__file__), "Data/dotastro_215153.dat"),
+        cfg.UPLOAD_FOLDER, cfg.FEATURES_FOLDER, "TEMP_TEST01",
+        None, None, False,
+        ['std_err'], False)
+    assert all("std_err" in d["features_dict"] for fname, d in res_dict.items())
+    assert all("ts_data" in d for fname, d in res_dict.items())
+    # os.remove(os.path.join(cfg.FEATURES_FOLDER, "TEMP_TEST01_features.csv"))
+    # os.remove(os.path.join(cfg.FEATURES_FOLDER, "TEMP_TEST01_classes.pkl"))
+    # os.remove(os.path.join(cfg.FEATURES_FOLDER,
+    #                        "TEMP_TEST01_features_with_classes.csv"))
+
+
+def test_create_feat_dict_and_list():
+    """Test create feat dict and list"""
+    feat_val_list, feat_dict = pred.create_feat_dict_and_list(
+        {"f1": 21.1, "f2": 3.1, "f3": 101.0},
+        ["f1", "f2", "f3"], ["f1", "f2", "f3"])
+    npt.assert_array_equal(feat_val_list, [21.1, 3.1, 101.0])
+    npt.assert_equal(feat_dict['f1'], 21.1)
+    feat_val_list, feat_dict = pred.create_feat_dict_and_list(
+        {"f1": 21.1, "f2": 3.1, "f3": 101.0},
+        ["f1", "f3"], ["f1", "f2", "f3"])
+    npt.assert_array_equal(feat_val_list, [21.1, 101.0])
+    npt.assert_equal(feat_dict['f1'], 21.1)
+    assert "f2" not in feat_dict
+
+
+def test_add_to_predict_results_dict():
+    """Test add data to predict results dict"""
+    results_dict = {}
+    pred.add_to_predict_results_dict(results_dict, [[0.2, 0.5, 0.3]], "TT.dat",
+                                     [1, 2, 3], {'f1': 2},
+                                     os.path.join(os.path.dirname(__file__),
+                                                  "Data"),
+                                     "test", 5)
+    npt.assert_array_equal(results_dict["TT.dat"]["ts_data"], [1, 2, 3])
```

Comment:
We may want to use assert_array_almost_equal, in case of minor differences between machines and platforms, but I'm going to merge for now.